lib/internal/http2/compat.js

```diff
@@ -519,6 +519,18 @@ class Http2ServerResponse extends Stream {
     return this[kStream].writableCorked;
   }
 
+  get writableHighWaterMark() {
+    return this[kStream].writableHighWaterMark;
+  }
+
+  get writableFinished() {
+    return this[kStream].writableFinished;
```

Comment:
I suspect this will need a special case and test for `state.headRequest`.

Reply:
Though I think `headRequest` is slightly broken or maybe I don't understand it. There is special handling for it in `end()` but not in `write()`? Not sure who is a good ping there. @jasnell?

Reply:
Issue: https://github.com/nodejs/node/issues/33543